test/hashing.jl

```diff
@@ -309,3 +309,13 @@ struct AUnionParam{T<:Union{Nothing,Float32,Float64}} end
 end
 
 @test Core.Compiler.is_foldable_nothrow(Base.infer_effects(hash, Tuple{Type{Int}, UInt}))
+
+@testset "issue #58386" begin
+    for T in [Int, UInt, Int128, UInt128]
```

Comment:
for 128-bit integers, `hash` and `hash_integer didn't coincide on 1.11.5 as well. So I would just restrict this to 64-bit ints